src/main/java/com/google/cloud/spanner/pgadapter/wireoutput/ParameterDescriptionResponse.java

```diff
@@ -26,16 +26,16 @@
   private static final int PARAMETER_NUMBER_LENGTH = 2;
   private static final int PARAMETER_LENGTH = 4;
 
-  private final List<Integer> parameters;
+  private final int[] parameters;
 
-  public ParameterDescriptionResponse(DataOutputStream output, List<Integer> parameters) {
+  public ParameterDescriptionResponse(DataOutputStream output, int[] parameters) {
```

Comment:
Why is this change required? Doesn't List provide more extensibility also safer?

Reply:
Not necessarily, both `List<Integer>` and `int[]` have their pros and cons. `List<Integer>` is easier to work with if the collection is changing. It is however less efficient, as it requires boxing and unboxing of the primitive values, and it can more easily lead to errors caused by `NullPointerException`s. The default value of an item in a `List<Integer>` is `null`. The default value in an `int[]` is `0`. The latter is also what the PG wire protocol uses as the default value for parameter types with an unknown value. The latter is the reason that I changed it, as it makes the code less bloated and better correspond with what the wire protocol uses.